src/compiler/binder.ts

```diff
@@ -3391,7 +3391,7 @@ function createBinder(): (file: SourceFile, options: CompilerOptions) => void {
 
     function bindSpecialPropertyAssignment(node: BindablePropertyAssignmentExpression) {
         // Class declarations in Typescript do not allow property declarations
-        const parentSymbol = lookupSymbolForPropertyAccess(node.left.expression, blockScopeContainer) || lookupSymbolForPropertyAccess(node.left.expression, container);

```

Comment:
now this recursive lookup is handled solely by `lookupSymbolForPropertyAccess`, so all its callers can benefit from the same logic